### PR TITLE
Add Coin change in  C code

### PR DESCRIPTION
**Pull request by Zuoxun**
Running results are as follows:
> 凑到目标金额所需的最少硬币数量为 2
凑到目标金额所需的最少硬币数量为 2

If this PR is related to coding or code translation, please fill out the checklist and paste the console outputs to the PR.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

File: codes/c/chapter_dynamic_programming/coin_change.c [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+#include "../utils/common.h"
+
+/* 求最小值 */
+int min(int a, int b) {
+    return a < b ? a : b;
+}
+
+/* 零钱兑换：动态规划 */
+int coinChangeDP(int coins[], int amt, int coins_size) {
```
Comment (Gonglja) [inline codes/c/chapter_dynamic_programming/coin_change.c:15]: `coinsSize`

File: codes/c/chapter_dynamic_programming/coin_change.c [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                // 不选和选硬币 i 这两种方案的较小值
+                dp[i][a] = min(dp[i - 1][a], dp[i][a - coins[i - 1]] + 1);
+            }
+        }
+    }
+    return dp[n][amt] != MAX ? dp[n][amt] : -1;
+}
+
+/* 零钱兑换：空间优化后的动态规划 */
+int coinChangeDPComp(int coins[], int amt, int coins_size) {
```
Comment (Gonglja) [inline codes/c/chapter_dynamic_programming/coin_change.c:41]: Same as above

File: codes/c/chapter_dynamic_programming/coin_change.c [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+            }
+        }
+    }
+    return dp[amt] != MAX ? dp[amt] : -1;
+}
+
+/* Driver code */
+int main() {
+    int coins[] = {1, 2, 5};
+    int coins_size = sizeof(coins) / sizeof(coins[0]);
```
Comment (Gonglja) [inline codes/c/chapter_dynamic_programming/coin_change.c:66]: Same as above

**Review by Gonglja — COMMENTED** [review]
These naming methods need to be changed, there is no problem with anything else.

**Review by Gonglja — COMMENTED** [review]
Can be merged @krahets

**Review by krahets — APPROVED** [review]

**Pull request merged**